docs/realm.js

```diff
@@ -288,13 +288,44 @@ Realm.defaultPath;
  *   object types in this Realm. **Required** when first creating a Realm at this `path`.
  * @property {number} [schemaVersion] - **Required** (and must be incremented) after
  *   changing the `schema`.
- * @property {Object} [sync] - Sync configuration parameters with the following 
+ * @property {Object} [sync] - Sync configuration parameters with the following
  *   child properties:
  *   - `user` - A `User` object obtained by calling `Realm.Sync.User.login`
  *   - `url` - A `string` which contains a valid Realm Sync url
  *   - `error` - A callback function which is called in error situations
  *   - `validate_ssl` - Indicating if SSL certificates must be validated
  *   - `ssl_trust_certificate_path` - A path where to find trusted SSL certificates
+ *   - `ssl_verify_callback` - A callback function used to accept or reject the server's
+ *      SSL certificate. ssl_verify_callback is called with an object of type
+ *      {
+ *        serverAddress: String
+ *        serverPort: Number
+ *        pemCertificate: String
+ *        preverifyOk: Number
+ *        depth: Number
+ *      }
+ *
+ *      The return value of ssl_verify_callback decides whether the certificate is accepted (true)
+ *      or rejected (false). The ssl_verify_callback function is only respected on platforms where
+ *      OpenSSL is used for the sync client, e.g. Linux.
+ *
+ *      When the sync client has received the server's certificate chain, it presents every certificate in
+ *      the chain to the ssl_verify_callback function. The depth argument specifies the position of the
+ *      certificate in the chain. depth = 0 represents the actual server certificate and the root
+ *      certificate has the highest depth. The certificate of highest depth will be presented first.
+ *
+ *      preverifyOk is 1 if OpenSSL has accepted the certificate, and
+ *      0 if OpenSSL has rejected it. It is generally safe to return true when preverifyOk is 1. If
+ *      preverifyOk is 0, an independent verification should be made.
+ *
+ *      One possible way of using the ssl_verify_callback function is to embed the known server certificate
+ *      in the client and accept the presented certificate if and only if it is equal to the known
+ *      certificate.
+ *
+ *      The purpose of ssl_verify_callback is to enable custom certificate pinning and to solve cases where
```

Comment:
Maybe a link to a general description of certificate pinning could help the reader.

Reply:
I changed "pinning" to "handling". 

I think, it would be somewhat out of place to link to some random page about certificate pinning. 

I just want to convey the message that this callback can be used to handle every type of
custom situation.

Feel free to come with suggestions if this is not good enough.